master/internal/api_experiment.go

```diff
@@ -1236,22 +1242,25 @@ func (a *apiServer) PatchExperiment(
 		}
 
 		if newResources != nil {
+			e, ok := experiment.ExperimentRegistry.Load(int(exp.Id))
+			if !ok {
+				return nil, api.NotFoundErrs("experiment", fmt.Sprint(exp.Id), true)
+			}
+
 			if newResources.MaxSlots != nil {
-				a.m.system.TellAt(actor.Addr("experiments", int(exp.Id)),
-					sproto.SetGroupMaxSlots{MaxSlots: ptrs.Ptr(int(*newResources.MaxSlots))})
+				msg := sproto.SetGroupMaxSlots{MaxSlots: ptrs.Ptr(int(*newResources.MaxSlots))}
+				go e.SetGroupMaxSlots(msg)
```

Comment:
i don't think there is any good reason for this `go ..`, let's just make it sync.